bin/urlsize.js

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env node
+'use strict';
+var urlsize = require('../'),
+    fs = require('fs'),
+    csv = require('fast-csv'),
+    through2 = require('through2'),
+    split2 = require('split2'),
+    streamify = require('stream-array'),
+    yargs = require('yargs')
+      .usage('$0 [ --stream [options] | --file <file> [options] | [options] <url> [<url>...] ]')
+      .describe('stream', 'stream URLs in, one per line')
+      .describe('file', 'read URLs from a text file (one per line)')
+        .alias('file', 'f')
+      .describe('desc', 'sort URLs by size descending (default: ascending)')
+        .boolean('desc')
+        .alias('desc', 'd')
+      .describe('csv', 'output comma-separated values')
+        .boolean('csv')
+        .alias('csv', 'c')
+      .describe('tsv', 'output tab-separated values')
+        .boolean('tsv')
+        .alias('tsv', 't')
+      .describe('out', 'write to this file')
+        .alias('out', 'o')
+      .describe('help', 'show this helpful message')
+      .describe('v', 'print more helpful messages to stderr')
+        .boolean('v')
+      .alias('help', 'h'),
+    options = yargs.argv,
+    help = options.help,
+    urls = options._;
+
+if (!options.stream
+    && !options.file
+    && !urls.length) {
+  help = true;
+} else if (urls[0] === '-') {
+  // console.warn('reading from stdin; you should probably be using --stream');
+  options.stream = true;
+  urls = [];
+}
+
+if (help) {
+  yargs.showHelp();
+  return process.exit(1);
+}
+
+var sizeOptions = {
+      unix: true,
+      sort: options.desc ? 'd' : 'a',
+      log: options.v
+        ? console.warn.bind(console)
+        : false
+    },
+    fopts = {
+      encoding: 'utf8'
+    };
+
+if (options.stream) {
```

Comment:
random feedback, one strategy I use to avoid this big `if else` chains is to put code like this in a function e.g. `function parseArguments() { // copy paste all the if else code here }` and then you can use the early return style, e.g.

```js
if (!foo) return bar
return foo
```

rather than

```js
if (foo) {
  return foo
} else {
  return bar
}
```